### PR TITLE
improve accessibility of sign-up modal

### LINKED ISSUE #7964: Improve accessibility of Sign Up Modal
**Please describe the desired behavior.**
Currently the Sign Up Modal has some elements whose accessibility can be improved by adding proper description.

These problems include the  missing alternative text for profile image,missing form label for the radio buttons, empty links for social icons, and orphaned form labelsas flagged by lighthouse and WAVE.

**Additional context (optional)**
WAVE report  

7962_before

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #7964 
Improved accessibility of Sign up modal by adding missing form links, fixing orphaned form links, adding aria-labels for social icons and add alternative text for profile photo.

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

## Screenshots 

Before   

7964_before

After  

7964_after

Thanks!

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@VladimirMikulic can you kindly review ? :v:

**Comment by VladimirMikulic** [comment]
@Tlazypanda great work! Would you be able to resolve these 7 errors and 19 warnings so its once and for all? Thanks.

**Comment by Tlazypanda** [comment]
@VladimirMikulic Hey! The rest of these errors are flagged from the rest of the page for which I will be opening different issues(for ex - header one) and of the warnings apart from the ones in this page the rest are alright since they are - 

- Tab index (which works alright) 
- Missing fieldset (cant be applied here since we require a legend text so applying it can mess up the styling)
- Underlined text (again part of the styling we can lose this one but want to know if everyone is okay with that)

**Comment by jywarren** [comment]
Perhaps we can ask some fellow Outreachy folks to just take a quick look over this so you all are connected a bit? I'll request a review from @shreyaa-s and @Shulammite-Aso -- we don't have to for every PR but it's nice to look out for one anothers' work every so often! And, feel free to help review some of their recent work too -- do you all have one that @Tlazypanda might be able to help look over? 

Thanks, all!!! 🎉 ❤️

File: app/views/users/_create_form.html.erb [hunk 1]
```diff
@@ -21,7 +21,7 @@
     <div class="row">
       <div class="col-md-6">
         <div class="form-group" id="username_div">
-          <label for="username"><%= translation('user_sessions.new.username') %></label>
+          <label for="username-signup"><%= translation('user_sessions.new.username') %></label>
```
Comment (jywarren) [inline app/views/users/_create_form.html.erb:24]: Just checking if this has affected our client-side validation at all? I would guess not but good to note this so people feel secure about it. Thank you!!!

File: app/views/users/_create_form.html.erb [hunk 2]
```diff
@@ -48,7 +49,7 @@
 
     <div class="form-group row">
       <div class="form-group col-md-6">
-        <label for="password"><%= translation('users._form.create_password') %></label>
+        <label for="password1"><%= translation('users._form.create_password') %></label>
```
Comment (jywarren) [inline app/views/users/_create_form.html.erb:52]: Same question about client-side validation!

File: app/views/users/_create_form.html.erb [hunk 3]
```diff
@@ -59,7 +60,7 @@
       </div>
 
       <div class="form-group col-md-6">
-        <label for="password_confirmation"><%= translation('users._form.password_confirmation') %></label>
+        <label for="password-confirmation"><%= translation('users._form.password_confirmation') %></label>
```
Comment (jywarren) [inline app/views/users/_create_form.html.erb:63]: and here!

File: app/views/users/_create_form.html.erb [hunk 4]
```diff
@@ -72,10 +73,11 @@
 
     <div class="form-group row">
       <div class="col-md-12">
-        <label for="user_bio"><%= translation('users._form.bio') %></label>
+        <label for="user-bio"><%= translation('users._form.bio') %></label>
```
Comment (jywarren) [inline app/views/users/_create_form.html.erb:76]: sorry, one more!

File: app/views/users/_spamaway.html.erb [hunk 5]
```diff
@@ -16,6 +14,7 @@
         <div class="btn-group d-flex" role="group">
           <% [0,1].each_with_index do |s, j| %><% statement = turingtest[i][s] %>
             <button type="button" class="col-5 btn btn-outline-secondary" style="font-size:3em;text-align:left;<% if i.odd? %> background:#eef;<% end %>" id="spamaway-<%= caller %>-<%= i.to_s + j.to_s %>">
+            <label id="selectAnimal" for="spamaway_statement_<%= caller %>_<%= i.to_s + j.to_s %>">
```
Comment (jywarren) [inline app/views/users/_spamaway.html.erb:17]: 🐯 🐘 🐻 😻

**Comment by shreyaa-s** [comment]
I think changing the labels shouldn't interfere with the validation since we use `getElementById` or `name` there. Looks good to me. :+1:

**Comment by Tlazypanda** [comment]
Hey @shreyaa-s @Shulammite-Aso @jywarren! Thanks for your reviews :heart: And I tested for the client side validation with a new user works fine :sweat_smile:

**Comment by jywarren** [comment]
Awesome! Please give this one last test at stable.publiclab.org -- thank you!

**Pull request merged**